projects/legacy/components/input-date/input-date.component.ts

```diff
@@ -174,9 +174,13 @@ export class TuiInputDateComponent
         }
 
         this.value =
-            value.length !== DATE_FILLER_LENGTH
+            value.length !== DATE_FILLER_LENGTH || TUI_LETTER_REGEXP.test(value)
                 ? null
                 : TuiDay.normalizeParse(value, this.dateFormat.mode);
+
+        if (TUI_LETTER_REGEXP.test(this.nativeValue)) {
+            this.nativeValue = maskitoTransform(this.nativeValue, this.computedMask);
```

Comment:
Why do we should use mask if textfield value contains any letters ? 🤔

Reply:
When mask object changes from mask for any value to mask for date (when `this.activeItem` return `false`), it applies only when user changed next value, but not current. If we don't trigger it manually, it would be:
"Until today" -> (press firstly backspace) -> "Until toda" -> (press backspace secondly) -> ""

But we need: "Until today" -> (press firstly backspace) -> "", then we need to dispatch `maskitoTransform` to apply updated mask

Reply:
Why not this?
```
this.nativeValue = ''
```

If we apply maskito date mask for any text value (not date string) – it will always return empty string